### ISSUE TITLE
TypeError: Cannot destructure property 'media' of 'theme.sanity' as it is undefined

### ISSUE THREAD

**Issue by danbeck79** [sanity-io/sanity-plugin-cloudinary#50]
Trying to implement the plugin..but neither of the following implementations work:

`cloudinaryAssetSourcePlugin `
`cloudinarySchemaPlugin`

I get attached error when trying to follow both implementations:

<img width="1187" alt="Screenshot 2023-07-19 at 14 28 18" src="https://github.com/sanity-io/sanity-plugin-cloudinary/assets/49392101/bc0f6f42-2220-4ddc-b21f-14739cae0176">

### Packages:

```shell
"@sanity/color": "^2.2.5",
"@sanity/icons": "^2.3.1",
"@sanity/image-url": "^1.0.2",
"@sanity/studio-secrets": "^2.0.2",
"@sanity/types": "^3.14.3",
"@sanity/ui": "^1.7.3",
"@sanity/vision": "^3.14.3",
"@sanity/webhook": "^2.0.0",
"sanity": "^3.14.3",
"sanity-plugin-cloudinary": "^1.1.1",
"sanity-plugin-documents-pane": "^2.1.0",
"sanity-plugin-iframe-pane": "^2.3.1",
"sanity-plugin-media": "^2.2.1",
```

### Sanity config plugins:

```javascript
export default defineConfig({
  ...
  plugins: [
    deskTool({
      defaultDocumentNode,
      structure,
    }),
    visionTool(),
    media(),
    cloudinarySchemaPlugin(),
  ],
 ...
})
``` 

### schema field definition:

```javascript
 defineField({
      name: 'media',
      title: 'Media',
      type: 'cloudinary.asset',
      description: 'This asset is served from Cloudinary',
    }),
``` 

Any ideas?

**Comment by rselimi** [comment]
@stipsan any chance for someone looking at this? Thanks.

**Comment by olbrown** [comment]
I'm seeing this issue on an existing site too. 

```
...
"@sanity/types": "3.16.7",
"@sanity/ui": "^1.8.2",
"sanity": "3.16.7",
"sanity-plugin-cloudinary": "~1.1.1",
...
```
🤷‍♂️

**Comment by olbrown** [comment]
> I'm seeing this issue on an existing site too.
> 
> ```
> ...
> "@sanity/types": "3.16.7",
> "@sanity/ui": "^1.8.2",
> "sanity": "3.16.7",
> "sanity-plugin-cloudinary": "~1.1.1",
> ...
> ```
> 
> 🤷‍♂️

In case this is useful to nayone else, I've managed to fix this by removing styled-components from my project.

**Comment by chalkstore** [comment]
Also experiencing this if anyone has a workaround

**Comment by musaa-muhsen** [comment]
Has anyone had any luck with this in October 2023?

**Comment by mittnavnermike** [comment]
So I got this working – I'm not sure what actually solved it, but this is what I did:

1. Remove `node_modules` and `pnpm-lock.yaml`
2. Install using node `npm install`
3. This gave me an error conflict with `styled-components` package
4. Removed the dependency in my `package.json`
5. Repeat step 1 and 2
6. `npm run dev` it was working for me at this point
7. Repeat step 1 and installed with pnpm

It's working now so I assume it's an issue with `styled-components` dependencies 🤷‍♂️

**Comment by musaa-muhsen** [comment]
Thank you for sharing your solution.

**Comment by WahidN** [comment]
Can this be resolved? I don't want to remove the `styled-components` dependency

**Comment by wg4568** [comment]
I'm seeing a similar issue on a clean sanity@latest starter install, this shows immediately on loading the web UI when using the schema plugin (`cloudinarySchemaPlugin`).

![Screenshot 2023-12-20 160002](https://github.com/sanity-io/sanity-plugin-cloudinary/assets/8494697/3ab25146-ff97-4140-bd40-8ab79015d9e4)

Definitely seems to be related to styled components. A warning is dropped in the console that reads:

`It looks like there are several instances of 'styled-components' initialized in this application. This may cause dynamic styles to not render properly, errors during the rehydration process, a missing theme prop, and makes your application bigger without good reason.`

```
"@sanity/vision": "^3.23.1",
"sanity": "^3.23.1",
"sanity-plugin-cloudinary": "^1.1.1",
"styled-components": "^6.1.2"
```

**Comment by laurosilvacom** [comment]
This issue might be related to what I'm experiencing https://github.com/sanity-io/sanity-plugin-cloudinary/issues/63

**Comment by wg4568** [comment]
Might be worth taking a look at the fix used in https://github.com/sanity-io/sanity-plugin-cloudinary/issues/42. Not sure why this works but it fixed my `theme.sanity is undefined` issue and appears to be working correctly.

```
    "@sanity/types": "3.14.2",
    "@sanity/vision": "3.14.2",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "sanity": "3.14.2",
    "sanity-plugin-cloudinary": "~1.1.1",
    "styled-components": "^6.1.3"
```

**Comment by bopontuskarlsson** [comment]
I have the same issue with the new install. Any ideas?

<img width="1311" alt="Skärmavbild 2024-01-08 kl  16 45 58" src="https://github.com/sanity-io/sanity-plugin-cloudinary/assets/1200021/ea0f4f44-6d9c-4efe-91a0-d7af71955792">

`"dependencies": {
    "@sanity/vision": "^3.23.4",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-is": "^18.2.0",
    "sanity": "^3.23.4",
    "sanity-plugin-cloudinary": "^1.1.1",
    "styled-components": "^6.1.7"
  },`

**Comment by emilsall** [comment]
I just got some help from Sanity support that suggested installing the plugin with:
`npm i sanity-plugin-cloudinary --legacy-peer-deps`
Which worked for me.

They also said they'll file an internal bug report for this issue 👍

**Comment by bobinska-dev** [comment]
Hello everyone! 
When these kind of issues arise (conflicts of dependencies), you can just make sure both are installed like this: 

`npm i sanity-plugin-cloudinary --legacy-peer-deps`

**Comment by laurosilvacom** [comment]
I'm having trouble installing the sanity-plugin-cloudinary package using pnpm. The equivalent command in pnpm for `npm i sanity-plugin-cloudinary --legacy-peer-deps` would be:

```
pnpm add sanity-plugin-cloudinary --strict-peer-dependencies=false
```

Unfortunately, this command does not work as expected. It seems there might be some compatibility issues or specific requirements of the `sanity-plugin-cloudinary` package that are not met when using pnpm.

@stipsan  could you please provide some guidance on how to install `sanity-plugin-cloudinary` with pnpm or how to suppress this issue? Any help would be greatly appreciated.

**Comment by djfarly** [comment]
struggled with this quite a lot (using pnpm) - for me it worked to add this to package.json:

```json
  "pnpm": {
    "overrides": {
      "styled-components": "6.1.8"
    }
  }
```

This results in this warning:

```
 WARN  Issues with peer dependencies found
studio
└─┬ sanity-plugin-cloudinary 1.1.1
  └─┬ @sanity/studio-secrets 2.0.2
    └── ✕ unmet peer styled-components@^5.3.0: found 6.1.8 in sanity-plugin-cloudinary
```
…but it seems to work.

Note: I'm using pnpm workspaces (monorepo) so I had to add this to the root package.json.

**Comment by laurosilvacom** [comment]
@djfarly, what version of `pnpm` are you using? I'm still trying to get this to work but my theory is that it's conflicting with a specific version of `pnpm`.

**Comment by djfarly** [comment]
@laurosilvacom was on `8.12.0`, but just updated to `8.15.1` which also seems to work fine.

**Comment by binoy14** [comment]
Hi all, sorry for the inconvenience we have released [v1.1.2](https://github.com/sanity-io/sanity-plugin-cloudinary/releases/tag/v1.1.2) that should fix the issue. If you are still having issues please open a new issue
